client/components/domains/register-domain-step/index.jsx

```diff
@@ -82,8 +82,8 @@ var RegisterDomainStep = React.createClass( {
 		}
 	},
 
-	componentDidUpdate: function( prevProps ) {
-		if ( this.props.selectedSite && this.props.selectedSite.domain !== prevProps.selectedSite.domain ) {
+	componentWillReceiveProps: function( nextProps ) {
+		if ( this.props.selectedSite && this.props.selectedSite.domain !== nextProps.selectedSite.domain ) {
 			this.setState( this.getInitialState() );
 			this.focusSearchCard();
 			this.fetchDefaultSuggestions();
```

Comment:
actually, here. This is what's using the old props and fetching suggestions for it.